extensions/notebook/src/book/bookEdit.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { JupyterBookSection } from '../contracts/content';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import * as vscode from 'vscode';
+import * as loc from '../common/localizedConstants';
+import { BookTocManager } from './bookTocManager';
+import { IBookUndoRedoElement } from './bookUndoRedoService';
+
+export interface undoRedoToc {
+	undo: JupyterBookSection[],
+	redo: JupyterBookSection[]
+}
+
+export class MoveBookTreeItem implements IBookUndoRedoElement {
+	label: string = loc.MoveBookTreeItem;
+
+	constructor(private bookTocManager: BookTocManager, private movedFiles: Map<string, string>, private tocFiles: Map<string, undoRedoToc>) {
+	}
+
+	async undo(): Promise<void> {
+		// restore toc files
+		for (const [tocPath, contents] of this.tocFiles.entries()) {
+			await this.applyTocChanges(tocPath, contents.undo);
```

Comment:
@barbaravaldez Just wanted to make sure you saw this. Did you test it out? This might be something we come back to after getting this initial implementation but I think it's probably going to end up being something we want to address before releasing to stable.

Reply:
I'll test undoing/redoing quickly. If I see any issues, I'll take a look at serializing the undo/redo operations.